libsolidity/ast/AST.cpp

```diff
@@ -757,6 +757,44 @@ FunctionCallAnnotation& FunctionCall::annotation() const
 	return initAnnotation<FunctionCallAnnotation>();
 }
 
+vector<ASTPointer<Expression const>> FunctionCall::sortedArguments() const
+{
+	// normal arguments
+	if (m_names.empty())
+		return arguments();
+
+	// named arguments
+	vector<ASTPointer<Expression const>> sorted;
+	FunctionTypePointer functionType;
+	if (*annotation().kind == FunctionCallKind::StructConstructorCall)
+	{
+		auto const& type = dynamic_cast<TypeType const&>(*m_expression->annotation().type);
+		auto const& structType = dynamic_cast<StructType const&>(*type.actualType());
+		functionType = structType.constructorType();
+	}
+	else
```

Comment:
How can it not be `StructConstructorCall` yet be named?

Reply:
A normal function call can also be named:
```
function f(uint x, uint y) {}
function g() { f({y: 2, x:3});
```

Reply:
https://github.com/ethereum/solidity/pull/10234/files?file-filters%5B%5D=.sol#diff-edbb3279b9dc84dda44fab7ee9ee066557c0fd656d43a892faceab192a60a68cR2

Reply:
But isn't `f({y: 2, x:3})` becoming `StructConstructorCall` in the ast?

Reply:
I don't think so